zipline/utils/input_validation.py

```diff
@@ -130,6 +133,19 @@ def ensure_timezone(func, argname, arg):
     )
 
 
+def expect_non_empty(func, argname, arg):
```

Comment:
So far all of the other functions here named `expect_<thing>` take **kwargs that get mapped to the arguments.  That seems like a worthwhile pattern to maintain.  Maybe this should be `expect_length_between` and take **lower_and_upper_bounds as pairs of ints?